File: CS/ParamNavigation/ucGeneral.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking2010.Views.WindowsUI;

namespace ParamNavigation {
    public partial class ucGeneral : UserControl, ISupportNavigation {
        WindowsUIView generalView;
        Page detailsPage;
        public ucGeneral() {
            InitializeComponent();
            employeesTableAdapter.Fill(nwindDataSet.Employees);
        }

        public void OnNavigatedFrom(INavigationArgs args) {
            args.Parameter = gridView1.GetDataRow(gridView1.FocusedRowHandle);
        }

        public void OnNavigatedTo(INavigationArgs args) {
            generalView = args.View;
            detailsPage = args.Parameter as Page;
        }

        private void gridView1_DoubleClick(object sender, EventArgs e) {
            generalView.Controller.Activate(detailsPage);
        }
    }
}

File: VB/ucGeneral.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking2010.Views.WindowsUI

Namespace ParamNavigation
	Public Partial Class ucGeneral
		Inherits UserControl
		Implements ISupportNavigation
		Private generalView As WindowsUIView
		Private detailsPage As Page
		Public Sub New()
			InitializeComponent()
			employeesTableAdapter.Fill(nwindDataSet.Employees)
		End Sub

        Public Sub OnNavigatedFrom(ByVal args As INavigationArgs) Implements ISupportNavigation.OnNavigatedFrom
            args.Parameter = gridView1.GetDataRow(gridView1.FocusedRowHandle)
        End Sub

        Public Sub OnNavigatedTo(ByVal args As INavigationArgs) Implements ISupportNavigation.OnNavigatedTo
            generalView = args.View
            detailsPage = TryCast(args.Parameter, Page)
        End Sub

		Private Sub gridView1_DoubleClick(ByVal sender As Object, ByVal e As EventArgs) Handles gridView1.DoubleClick
			generalView.Controller.Activate(detailsPage)
		End Sub
	End Class
End Namespace
